include/cuco/detail/static_map.inl

```diff
@@ -402,7 +403,7 @@ __device__ bool static_map<Key, Value, Scope, Allocator>::device_mutable_view::i
         }
 
         if constexpr (not cuco::detail::is_packable<value_type>()) {
-#if __CUDA_ARCH__ < 700
+#if !defined(CUCO_HAS_INDEPENDENT_THREADS)
```

Comment:
This is bad. Since this macro is now dependent on the minimum architecture instead of the current arch, this would disable back-to-back CAS even on Volta+.
We need to differentiate the macro into `CUCO_CURRENT_ARCH_HAS_INDEPENDENT_THREADS` for device code and `CUCO_ALL_ARCHS_HAVE_INDEPENDENT_THREADS` for host code. Or... we roll back to using `__CUDA_ARCH__` here.

Reply:
> This is bad. Since this macro is now dependent on the minimum architecture instead of the current arch

I'm inclined to leave it as it is now.

Reply:
You mean rolling back to using `__CUDA_ARCH__`? Yeah this should totally be fine.